server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -606,7 +613,25 @@ public ConsumerDTO create(
         @QueryParam("identity_cert_creation") @DefaultValue("true") boolean identityCertCreation)
         throws BadRequestException {
 
-        Consumer consumer = new Consumer();
+        // fix for duplicate hypervisor/consumer problem
+        Consumer consumer = null;
+        if (ownerKey != null && dto.getFact("system_uuid") != null &&
+            !"true".equalsIgnoreCase(dto.getFact("virt.is_guest"))) {
+            Owner owner = ownerCurator.lookupByKey(ownerKey);
+            if (owner != null) {
+                consumer = consumerCurator.getHypervisor(dto.getFact("system_uuid"), owner);
+                if (consumer != null) {
+                    consumer.setIdCert(generateIdCert(consumer, false));
+                    this.updateConsumer(consumer.getUuid(), dto, principal);
+                    return translator.translate(
+                        consumerCurator.findByUuid(consumer.getUuid()), ConsumerDTO.class);
```

Comment:
Do we need to do another lookup here? We have the consumer already, is there a reason we're not just translating it straight away?